DEVELOPING.md

```diff
@@ -0,0 +1,129 @@
+# Developing Liquibase
+
+## Overview
+
+Liquibase is implemented in Java, and uses Maven as it's build system. 
+
+The code structure follows the standard Maven directory structure divided into modules, with `liquibase-core` being the most central module.
+
+## Testing
+
+While there are still older test classes written in JUnit, tests should be written in (or if possible migrated to) [Spock](https://spockframework.org/).
+
+### Integration Testing
+
+**NOTE: THIS DESCRIBES AN ALPHA SYSTEM WHICH IS LIKELY TO CHANGE. CHECK BACK REGULARLY**
+
+Because Liquibase interacts with many different databases, it is important that we automate the testing of those interactions.
+
+### Integration Testing Environments
+
+In order to write a test against an external database, you must first have a database to test against. 
+Liquibase handles this for you both to keep you focused on test writing, and also to provide standardization and repeatability to the tests.
+
+#### Referencing a test system from your tests  
+
+Within your tests, you create a reference to the system you are going to test against by calling `testSystem = Scope.getCurrentScope().getSingleton(TestSystemFactory.class).getTestSystem("h2")` where the argument to `getTestSystem()` is the key for a configured system.
+The default  test systems are configured in [liquibase.sdk.yaml](liquibase-extension-testing/src/main/resources/liquibase.sdk.yaml). 
+When that line in your test is hit, the TestSystemFactory will use a running system if available, or start a new system if needed. 
+Any newly-started systems will be available for the lifetime of **all** the tests to be shared between them and be shut down at the end of the entire run.
+
+The base TestSystem class is designed to work for more than just Databases, so if you know you are connecting to a database you will want to cast it to a DatabaseTestSystem in order to call methods like `getConnection()` for use in connecting to it. 
+For example:
+```java
+@Shared
+private DatabaseTestSystem mysql = 
+    (DatabaseTestSystem) Scope.getCurrentScope().getSingleton(TestSystemFactory.class).getTestSystem("mysql")
+```
+
+When TestSystems need to be started, Liquibase generally relies on Docker instances. For time when docker images are not available or are unneeded, they can also connect to an external database via a `url` setting. 
+
+When using docker instances, if we want to add support for a new database we want to test against, we need to create a new test class for which we will need to extend from `DatabaseTestSystem` and override
```

Comment:
Minor edit to line 41 to tighten up the wording a bit:

"... we need to create a new test class extended from `DatabaseTestSystem` and override ..."